### PR TITLE
7. Media

### PR DESCRIPTION
**Pull request by amywieliczka**
This attempts to create an abstraction layer for the Solr side of things and only includes some marginal level of Rikolti adaptation - see #295 for a more complete Rikolti adaptation of media handling

### CODE REVIEW

File: calisphere/record.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        self.repositories = [
+            Repository(repo_id, index) 
+            for repo_id in self.indexed_record.get('repository_ids')
+        ]
+
+    def is_hosted(self):
+        if self.index == 'solr':
+            return bool(self.indexed_record.get('structmap_url'))
+        elif self.index == 'es':
+            return bool(self.indexed_record.get('media'))
```
Comment (barbarahui) [inline calisphere/record.py:131]:
```suggestion
            return bool(self.indexed_record.get('mapper_type') == 'nuxeo.nuxeo')
```

File: calisphere/record.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
+from copy import deepcopy
+from .es_cache_retry import json_loads_url
+from typing import Any, Dict
+from django.conf import settings
+from django.http import Http404
+from .collection_views import Collection
+from .institution_views import Repository
+
+def get_iiif(media_id) -> Dict[str, Any]:
+    iiif_url = f"{settings.UCLDC_IIIF}{media_id}/info.json"
```
Comment (barbarahui) [inline calisphere/record.py:10]:
```suggestion
    iiif_url = f"{settings.UCLDC_IIIF}{urllib.parse.quote(media_key)}/info.json"
```

**Comment by barbarahui** [comment]
@amywieliczka I looked through the 7 PRs that culmulatively end with this one. PRs 1-6 look good to me. Your approach here also makes a lot of sense to me.

I just added a couple of 1-line code suggestions to `calisphere/record.py` that reflect some work I did that isn't incorporated here. (I know this is a draft; just didn't want it to get lost):

1. The existence of `media` isn't a reliable indicator of whether or not a record is hosted; some complex objects don't have `media` at the parent level.
2. The `iiif_url` now needs to be url encoded as we are storing files on s3 by filename, not md5 hash. Some of the filenames contain special characters.

File: calisphere/record.py (hunk 1 above)
Comment (amywieliczka) [reply calisphere/record.py:131]: I made this change here:
https://github.com/ucldc/public_interface/pull/295/files#diff-070924927e1cadaac0c17b98fb7f54f4997438cdac87c01c696bfcac3f2b6e80R152

File: calisphere/record.py (hunk 2 above)
Comment (amywieliczka) [reply calisphere/record.py:10]: I made this change here as part of #295 here: https://github.com/ucldc/public_interface/pull/295/commits/86c31c69cd177b0589db9ed22eed85d2b6f9d696

**Pull request merged**